core/src/test/java/org/infinispan/distexec/DistributedExecutorBadResponseFailoverTest.java

```diff
@@ -0,0 +1,131 @@
+import org.infinispan.test.TestingUtil;
+import org.testng.AssertJUnit;
+import org.testng.annotations.Test;
+
+/**
+ * Tests dist.exec failover due to CacheNotFoundResponse or any other non SuccessfulResponse response.
+ *
+ * @author Vladimir Blagojevic
+ * @since 7.1
+ */
+@Test(groups = "functional", testName = "distexec.DistributedExecutorBadResponseFailoverTest")
+public class DistributedExecutorBadResponseFailoverTest extends MultipleCacheManagersTest {
+
+   @Override
+   protected void createCacheManagers() throws Throwable {
+      ConfigurationBuilder builder = getDefaultClusteredCacheConfig(CacheMode.DIST_SYNC, false);
+      addClusterEnabledCacheManager(builder);
+      addClusterEnabledCacheManager(builder);
+      addClusterEnabledCacheManager(builder);
+
+      // this is the node sending DistributedExecuteCommand
+      final EmbeddedCacheManager cacheManager1 = manager(0);
+      TestingUtil.wrapGlobalComponent(cacheManager1, Transport.class,
+            new TestingUtil.WrapFactory<Transport, Transport, CacheContainer>() {
+
+               @Override
+               public Transport wrap(CacheContainer wrapOn, Transport current) {
+                  return new CacheNotFoundResponseTransport();
+               }
+            }, true);
+
+      waitForClusterToForm(cacheName());
+   }
+
+   protected String cacheName() {
+      return "DistributedExecutorBadResponseFailoverTest";
+   }
+
+   public void testBasicTargetRemoteDistributedCallable() throws Exception {
+      long taskTimeout = TimeUnit.SECONDS.toMillis(15);
+      EmbeddedCacheManager cacheManager1 = manager(0);
+      final EmbeddedCacheManager cacheManager2 = manager(1);
+
+      Cache<Object, Object> cache1 = cacheManager1.getCache();
+      Cache<Object, Object> cache2 = cacheManager2.getCache();
+      DistributedExecutorService des = null;
+
+      try {
+         des = new DefaultExecutorService(cache1);
+         Address target = cache2.getAdvancedCache().getRpcManager().getAddress();
+
+         DistributedTaskBuilder<Integer> builder = des.createDistributedTaskBuilder(new SimpleCallable())
+               .failoverPolicy(DefaultExecutorService.RANDOM_NODE_FAILOVER)
+               .timeout(taskTimeout, TimeUnit.MILLISECONDS);
+
+         Future<Integer> future = des.submit(target, builder.build());
+         AssertJUnit.assertEquals((Integer) 1, future.get());
+      } catch (Exception ex) {
+         AssertJUnit.fail("Task did not failover properly " + ex);
+      } finally {
+         des.shutdown();
+      }
+   }
+
+   static class SimpleCallable implements Callable<Integer>, Serializable {
+
+      /**
+       *
+       */
+      private static final long serialVersionUID = -3130274337449595197L;
+
+      public SimpleCallable() {
+      }
+
+      @Override
+      public Integer call() throws Exception {
+         return 1;
+      }
+   }
+
+   private static class CacheNotFoundResponseTransport extends JGroupsTransport {
```

Comment:
I think you should delegate to the existing transport instead of creating a new one. The test works fine now because the transport is only started on the first `getCache()` call (from `waitForClusterToForm()`), but if we ever change that we'll leak a JChannel.